app/models/exchange_variant.rb

```diff
@@ -3,4 +3,9 @@
 class ExchangeVariant < ApplicationRecord
   belongs_to :exchange
   belongs_to :variant, class_name: 'Spree::Variant'
+  after_destroy :destroy_related_outgoing_variants
+
+  def destroy_related_outgoing_variants
+    VariantDeleter.new.destroy_related_outgoing_variants(variant_id, exchange.receiver)
```

Comment:
I think that instead of the receiver, we should go by the order cycle. In theory, this should work fine but I keep wondering if there may be a case in the future where we delete an incoming exchange to one order cycle and not want to touch the exchange variants of other order cycles. So this ExchangeVariant has an OrderCycle through the Exchange. And we want to delete all ExchangeVariants of this Spree::Variant within this order cycle, if the destroyed exchange was an incoming one. Does that make sense?

Reply:
> In theory, this should work fine but I keep wondering if there may be a case in the future where we delete an incoming exchange to one order cycle and not want to touch the exchange variants of other order cycles.

I got a little confused here. 'this' is referring to the current implementation and not your suggested implementation of using the order cycle to destroy variants, right?

Reply:
Yes, that's right.